paasta_tools/marathon_dashboard.py

```diff
@@ -77,33 +79,48 @@ def create_marathon_dashboard(
     marathon_servers = get_marathon_servers(system_paasta_config=system_paasta_config)
     if marathon_clients is None:
         marathon_clients = get_marathon_clients(marathon_servers=marathon_servers, cached=False)
-    for service_instance in instances:
-        service: str = service_instance[0]
-        instance: str = service_instance[1]
-        service_config: MarathonServiceConfig = load_marathon_service_config(
+
+    dashboard_links: Dict = system_paasta_config.get_dashboard_links()
+    marathon_links = dashboard_links.get(cluster, {}).get('Marathon RO')
+
+    # e.g. 'http://10.64.97.75:5052': 'http://marathon-norcal-prod.yelpcorp.com'
+    shard_url_to_marathon_link_dict: Dict[str, str] = {}
+    if isinstance(marathon_links, list):
+        # Sanity check and log error if necessary
+        if len(marathon_links) != len(marathon_servers.current):
+            log.error('len(marathon_links) != len(marathon_servers.current). This may be a cause of concern')
+        for shard_number, shard in enumerate(marathon_servers.current):
+            shard_url_to_marathon_link_dict[shard.url[0]] = marathon_links[shard_number]
+    elif isinstance(marathon_links, str):
+        # In this case, the shard url will be the same for every service instance
```

Comment:
I agree. This was for backwards compatibility during the transition.

If you want to convert them all to arrays beforehand in puppet (`git grep Marathon RO`) it would simplify your code.